code/controllers/subsystem/shuttle.dm

```diff
@@ -594,6 +594,15 @@ SUBSYSTEM_DEF(shuttle)
 			return TRUE
 
 /datum/controller/subsystem/shuttle/proc/get_containing_shuttle(atom/A)
-	for(var/obj/docking_port/mobile/M in mobile)
-		if(M.is_in_shuttle_bounds(A))
-			return M
+	var/list/mobile_cache = mobile
+	for(var/i in 1 to mobile_cache)
+		var/obj/docking_port/port = mobile_cache[1]
```

Comment:
Shouldn't this be mobile_cache[i]?

Reply:
woops yeah I forgot to include the last batch of changes in the commit